debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSnapshotChangeEventSource.java

```diff
@@ -293,6 +293,12 @@ protected void setSnapshotTransactionIsolationLevel(boolean isOnDemand) throws S
         } else {
             LOGGER.info("Skipping setting snapshot time, snapshot data will not be consistent");
         }
+
+        // Regardless of whether consistent snapshot is enabled or not, we need to set the
+        // transaction isolation level.
+        String transactionIsolationLevelStatement = "SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE;";
+        LOGGER.info("Setting transaction isolation levels with statement {}", transactionIsolationLevelStatement);
+        jdbcConnection.executeWithoutCommitting(transactionIsolationLevelStatement);
```

Comment:
On line 292, we run the following in executeWithoutCommitting: `SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE;` If consistent snapshot is enabled then it will lead to execution of this statement twice. I think we can get rid of the if-else above.

Reply:
If consistent snapshot is enabled, this statement was never getting executed and thus we need to execute this regardless of consistent snapshot being enabled or not.